toolkit/smart-contracts/offchain/src/register.rs

```diff
@@ -23,8 +23,15 @@ use partner_chains_plutus_data::registered_candidates::{
 };
 use sidechain_domain::*;
 
+/// Trait for [register], to make it mockable.
 pub trait Register {
 	#[allow(async_fn_in_trait)]
+	/// This function submits a transaction to register a registered candidate.
+	/// Arguments:
+	///  - `retries`: Configuration for the retry logic of the transaction.
+	///  - `genesis_utxo`: UTxO identifying the Partner Chain.
+	///  - `candidate_registration`: [CandidateRegistration] registration data.
+	///  - `payment_signing_key`: Signing key of a governance authority member.
```

Comment:
Register and deregister has nothing to do with governance. This is payment key of the registering user.